src/Tempest/Http/src/Routing/Construction/MarkedRoute.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Http\Routing\Construction;
+
+use Tempest\Http\Route;
+
+final readonly class MarkedRoute
```

Comment:
I needed a clean way to carry the route `mark` used in the matching regex and the route into the tree.

Reply:
Fair enough, though isn't this a code smell indicating that we're using route mark in two different places? My assumption would be that the tree should manage the mark internally as well, and that no other place should know about it. Maybe this circles back to my question about whether the dynamic and static routes arrays in `RouteConfig` are still necessary? 

Again, not saying we should refactor it in this PR, but worth discussing imo

Reply:
Well, yes and no. 

Yes, currently "Route Construction" is done into the RouteConfig, and maybe it shouldn't . RouteConfig could also be used to only carry the resulting regexes and routes array, while not caring about the `MarkedRoute` concept and `RoutingTree`. But these regexes need to be build somewhere. If there was a function part of the `RouteDiscovery` class which was called after discovery was done. I could separate the discovery phase from the matching phase.


The  dynamic route array is a result of the routing tree, the routing tree is used to build up the matching regexes, not the routing itself. The static routes are not added to the routing tree, the current implementation is the most efficient (and easiest) one.

Reply:
Ok, let's see whether `KernelEvent::BOOTED` I mentioned in the other comment helps you. Apart from that, we don't need to worry about it in this PR. We can further improve after this has been merged.